app/src/ui/changes/filter-changes-list.tsx

```diff
@@ -1149,39 +1197,254 @@ export class FilterChangesList extends React.Component<
+          className="changes-list-check-all"
         />
 
-        <Button
-          onClick={this.onFilterToIncludedInCommit}
-          className={classNames({
-            'included-in-commit-filter-on': this.state.filterToIncludedCommit,
-          })}
-          ariaLabel={toBeCommittedFilterText}
-          tooltip={toBeCommittedFilterText}
-        >
-          <Octicon symbol={octicons.filter} />
-        </Button>
-        <label id="changes-list-check-all-label"> {filesDescription}</label>
+        <label id="changes-list-check-all-label">{visibleFiles} files</label>
+        <span className="spacer"></span>
+        {showHiddenCheckedFilesWarning ? (
+          <span>
+            <Button
+              className="filter-button hidden-files-warning"
+              ariaExpanded={this.state.isFilterOptionsOpen}
+              ariaLabel="Open Files Hidden by Filter Warning"
+              tooltip="Open Files Hidden by Filter Warning"
+              onClick={this.openFilterWarning}
+              onButtonRef={this.onFilterWarningButtonRef}
+            >
+              <Octicon symbol={octicons.alert} />
+            </Button>
+            {this.state.isFilesWarningOpen && this.renderFilterWarning()}
+          </span>
+        ) : null}
+
+        {this.state.filterToIncludedCommit ? (
+          <Button
+            className="filter-button remove-filter"
+            onClick={this.clearNonTextFilters}
+            ariaLabel="Clear 1 additional filter(s)"
+            tooltip="Clear 1 additional filter(s)"
+          >
+            <Octicon symbol={octicons.x} />
+          </Button>
+        ) : null}
+
+        <span>
+          {this.renderBetaPill()}
+
+          <Button
+            className={classNames('filter-button', {
+              active: this.state.filterToIncludedCommit,
+            })}
+            onClick={this.openFilterOptions}
+            ariaExpanded={this.state.isFilterOptionsOpen}
+            onButtonRef={this.onFilterOptionsButtonRef}
+            tooltip={
+              !this.props.canFilterChanges
+                ? 'Enable Filter (Beta)'
+                : 'Filter Options'
+            }
+            ariaLabel={
+              !this.props.canFilterChanges
+                ? 'Enable Filter (Beta)'
+                : 'Filter Options'
+            }
+          >
+            <Octicon symbol={octicons.filter} />
+            {this.props.canFilterChanges ? (
+              <Octicon symbol={octicons.triangleDown} />
+            ) : (
+              <Octicon symbol={octicons.triangleUp} />
+            )}
+          </Button>
+          {this.state.isFilterOptionsOpen && this.renderFilterOptions()}
+        </span>
       </div>
     )
   }
 
+  private renderBetaPill = () => {
+    if (this.props.canFilterChanges) {
+      return
+    }
+
+    return (
+      <span className="beta-pill">
+        <span>
+          <span>Beta</span>
+        </span>
+      </span>
+    )
```

Comment:
Component opportunity in the future? 🤔 

Separately, the pill looks larger than I would've expected according to the video. Should/Could it be smaller?

Reply:
Ha - yeah - I want sure if I should drop to xs font- happy to try that

Reply:
There could be some component or prior art somewhere.. but search I didn't find it.. so a component sounds lovely to me